extension/app/ts/background/background.ts

```diff
@@ -94,20 +104,29 @@ export async function refreshConfirmTransactionSimulation(
 	transactionToSimulate: EthereumUnsignedTransaction,
 	website: Website,
 ) {
-	if (simulator === undefined) return undefined
+	const info = {
+		requestId: requestId,
+		transactionToSimulate: transactionToSimulate,
+		simulationMode: simulationMode,
+		activeAddress: activeAddress,
+		signerName: await getSignerName(),
+		website: website,
+	}
+	if (simulator === undefined) return { method: 'popup_confirm_transaction_simulation_failed' as const, data: info }
 	sendPopupMessageToOpenWindows({ method: 'popup_confirm_transaction_simulation_started' })
-	const simulationStateWithNewTransaction = await appendTransaction(ethereumClientService, copySimulationState(simulationState), { transaction: transactionToSimulate, website: website })
-	return {
-		method: 'popup_confirm_transaction_simulation_state_changed' as const,
-		data: {
-			requestId: requestId,
-			transactionToSimulate: transactionToSimulate,
-			simulationMode: simulationMode,
-			...await visualizeSimulatorState(simulationStateWithNewTransaction, simulator),
-			activeAddress: activeAddress,
-			signerName: await getSignerName(),
-			website: website,
+	try {
+		const simulationStateWithNewTransaction = await appendTransaction(ethereumClientService, copySimulationState(simulationState), { transaction: transactionToSimulate, website: website })
+		return {
+			method: 'popup_confirm_transaction_simulation_state_changed' as const,
+			data: { ...info, ...await visualizeSimulatorState(simulationStateWithNewTransaction, simulator) }
 		}
+	} catch(error) {
+		if (error instanceof Error) {
+			if (isFailedToFetchError(error)) {
+				return { method: 'popup_confirm_transaction_simulation_failed' as const, data: info }
+			}
+		}
+		throw error
```

Comment:
Consider inverting the conditional here with guard clauses?
```suggestion
		if (!(error instanceof Error)) throw error
		if (!isFailedToFetchError(error)) throw error
		return { method: 'popup_confirm_transaction_simulation_failed' as const, data: info }
```